### PR TITLE
fix: preserve synthetic memorySessionId for stateless providers across worker restarts

### PR DESCRIPTION
**Pull request by Scheevel**
## Problem

When the worker restarts and recovers sessions via `SessionManager.initializeSession()`,
all `memory_session_id` values are unconditionally set to `null` (Issue #817 fix).

This is correct for SDK UUIDs — the SDK conversation context is lost on restart, so
the old ID causes "No conversation found" crashes on resume.

However, stateless providers (OpenRouter, Gemini) generate synthetic IDs with format
`{provider}-{contentSessionId}-{Date.now()}` that have no server-side state. These
remain valid across restarts and discarding them forces unnecessary ID regeneration.

## Fix

Replaces the unconditional discard in `SessionManager.ts` with conditional logic:

- **Synthetic IDs** (`openrouter-*`, `gemini-*`): Preserved from database
- **SDK UUIDs**: Discarded (existing Issue #817 behavior unchanged)

Detection uses strict regex validation matching the exact format generated by
`OpenRouterAgent.ts` and `GeminiAgent.ts`:

```
^(openrouter|gemini)-([0-9a-f]{8}-[0-9a-f]{4}-...-[0-9a-f]{12})-(\d+)$
```

## Changes

- `src/services/worker/SessionManager.ts` — `STATELESS_PROVIDERS` constant,
  `isSyntheticMemorySessionId()` method, conditional preserve/discard logic
- `tests/session_manager_synthetic_ids.test.ts` — 11 test cases covering
  preserve, discard, edge cases, and regex validation

## Related

- Issue #817 — original unconditional discard fix
- PR #615 — synthetic ID generation in OpenRouterAgent/GeminiAgent

### CODE REVIEW

File: src/services/worker/SessionManager.ts [hunk 1]
```diff
@@ -10,6 +10,12 @@
 
 import { EventEmitter } from 'events';
 import { DatabaseManager } from './DatabaseManager.js';
+
+// Stateless provider identifiers for synthetic ID generation
+export const STATELESS_PROVIDERS = {
+  GEMINI: 'gemini',
+  OPENROUTER: 'openrouter'
+} as const;
 import { logger } from '../../utils/logger.js';
```
[the remark this answers is not in the document]
Comment (Scheevel) [reply src/services/worker/SessionManager.ts]: Acknowledged: Fixed in updated commit and moved STATELESS_PROVIDERS below all imports.

File: src/services/worker/SessionManager.ts [hunk 2]
```diff
@@ -45,6 +51,23 @@ export class SessionManager {
...
+   * contentSessionId must be valid UUID format; suffix is Date.now() timestamp (digits)
+   */
+  private isSyntheticMemorySessionId(memorySessionId: string): boolean {
+    // Strict pattern: provider-(UUID)-(timestamp)
+    // UUID = contentSessionId from Claude Code session
+    // Timestamp = Date.now() at generation time (13+ digit number)
+    const uuidPattern = '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}';
+    const providers = Object.values(STATELESS_PROVIDERS).join('|');
+    const syntheticPattern = new RegExp(`^(${providers})-(${uuidPattern})-(\\d+)$`, 'i');
+    return syntheticPattern.test(memorySessionId);
```
[the remark this answers is not in the document]
Comment (Scheevel) [reply src/services/worker/SessionManager.ts:68]: Disagree: The \\d+ in a template literal correctly produces \d+ in the RegExp constructor string.
The double backslash escapes to single backslash. 
This was validated against real upstream synthetic IDs.
Test module included, and these tests pass.

**Comment by thedotmack** [comment]
Closed by @thedotmack. Please reopen if still relevant. Thank you for your contribution!
